op-chain-ops/addresses/roles.go

```diff
@@ -28,3 +36,21 @@ type OpChainFaultProofsRoles struct {
 	Proposer   common.Address
 	Challenger common.Address
 }
+
+var ErrSuperchainRoleZeroAddress = errors.New("SuperchainRole is set to zero address")
+
+func (s *SuperchainRoles) CheckNoZeroAddresses() error {
```

Comment:
What does this function do exactly? Because it looks to me like it has a known type (SuperchainRoles), then it reflects on that type and makes sure all fields are addresses, it does not seem to check for any zeros

Reply:
It just returns an error if any of the fields within the struct are set to zero (i.e. uninitialized). I added a couple tests here to confirm its working properly https://github.com/ethereum-optimism/optimism/pull/15519/commits/44287a044e3986daffcd6e44dc53bb59761be580